examples/cms-sanity/README.md

```diff
@@ -0,0 +1,140 @@
+
+Download the example:
+
+```bash
+curl https://codeload.github.com/zeit/next.js/tar.gz/canary | tar -xz --strip=2 next.js-canary/examples/cms-sanity
+cd cms-sanity
+```
+
+## Configuration
+
+### Step 1. Create an account and a project on Sanity
+
+First, [create an account on Sanity](https://sanity.io).
+
+After creating an account, install the Sanity cli from npm `npm i -g @sanity/cli`.
+
+### Step 2. Create a new Sanity project
+
+In a separate folder run `sanity init` to initialize a new studio project.
+
+This will be where we manage our data.
+
+When going through the init phase make sure to select **Yes** to the **Use the default dataset configuration** step and select **Clean project with no predefined schemas** for the **Select project template** step.
+
+After initializing, get the `projectId` value from the `sanity.json` file that was generated for the next step.
+
+### Step 3. Set up environment variables
+
+First, copy the `.env.example` file in this directory to `.env` (which will be ignored by Git):
+
+```bash
+cp .env.example .env
+```
+
+Then set each variable on `.env`:
+
+- `NEXT_EXAMPLE_CMS_SANITY_PREVIEW_SECRET` can be any random string (but avoid spaces), like `MY_SECRET` - this is used for [the Preview Mode](/docs/advanced-features/preview-mode.md).
+- `SANITY_PROJECT_ID` the `projectId` from the previous step
+
+Your `.env` file should look like this:
+
+```bash
+NEXT_EXAMPLE_CMS_SANITY_PREVIEW_SECRET=...
+SANITY_PROJECT_ID=...
+```
+
+### Step 4. Copy the schema file
+
+After initializing your Sanity studio project there should be a `schemas` folder.
+
+To get up and running we want to copy the schema file from [`./schemas/schema.js`](./schemas/schema.js) to the studio project to use for creating/editing content.
+
+### Step 5. Populate Content
+
+To add some content go to your Sanity studio project where we added the schema file and run `sanity start`.
+
+After the project has started and you have navigated to the URL given in the terminal, select **Author** and create a new record.
+
+- You just need **1 Author record**.
+- Use dummy data for the text.
+- For the image, you can download one from [Unsplash](https://unsplash.com/).
+
+Next, select **Post** and create a new record.
+
+- We recommend creating at least **2 Post records**.
+- Use dummy data for the text.
+- You can write markdown for the **Content** field.
+- For the images, you can download ones from [Unsplash](https://unsplash.com/).
+- Pick the **Author** you created earlier.
+
+**Important:** For each post record, you need to click **Publish** after saving. If not, the post will be in the draft state.
+
+### Step 6. Run Next.js in development mode
+
+```bash
+npm install
+npm run dev
+
+# or
+
+yarn install
+yarn dev
+```
+
+Your blog should be up and running on [http://localhost:3000](http://localhost:3000)! If it doesn't work, post on [GitHub discussions](https://github.com/zeit/next.js/discussions).
+
+### Step 7. Try preview mode
+
```

Comment:
To do preview, we need to have the user go through the instructions here:

https://www.sanity.io/docs/preview-content-on-site

and instead of `https://my-site.com/preview/${document._id}`, have them use ``http://localhost:3000/api/preview?secret=...&slug=${document.slug}`

Then they need to generate a read-only token here: https://manage.sanity.io/ → Select Settings → API → Tokens → Read

Then save that to `.env`.

Reply:
Added this to the step 👍